src/components/button/button.spec.js

```diff
@@ -66,12 +66,51 @@ describe('md-button', function() {
     expect(button[0]).not.toHaveClass('md-focused');
   }));
 
-  it('should set focus state on focus and remove on blur', inject(function ($compile, $rootScope){
+  it('should apply focus effect with keyboard interaction', inject(function ($compile, $rootScope){
     var button = $compile('<md-button>')($rootScope.$new());
+    var body = angular.element(document.body);
+
     $rootScope.$apply();
+
+    // Fake a keyboard interaction for the $mdInteraction service.
+    body.triggerHandler('keydown');
     button.triggerHandler('focus');
+
     expect(button[0]).toHaveClass('md-focused');
+
     button.triggerHandler('blur');
+
+    expect(button[0]).not.toHaveClass('md-focused');
+  }));
+
+  it('should apply focus effect when programmatically focusing', inject(function ($compile, $rootScope){
+    var button = $compile('<md-button>')($rootScope.$new());
+
+    $rootScope.$apply();
+
+    button.triggerHandler('focus');
+
+    expect(button[0]).toHaveClass('md-focused');
```

Comment:
That's why I didn't make it a blocking request. It's basically the same, it just looks nicer.